docs/development.rst

```diff
@@ -2,9 +2,177 @@
     :start-after: sec-begin-development
     :end-before: sec-end-development
 
+.. _writing-adapters:
+
 Writing a model adapter
 =======================
 
+Creating an :class:`~openscm.adapter.Adapter` subclass
+******************************************************
+
+- Create your adapter source file in ``openscm/adapters/``, e.g.
+  ``myadapter.py``
+
+- If your adapter needs additional dependencies add them to the
+  ``requirements_extras`` dictionary in ``setup.py`` (see comment
+  there).
+
+- Subclass the :class:`openscm.adapter.Adapter` class:
+
+  .. code:: python
+
+      from ..adapter import Adapter
+
+      class MyAdapter(Adapter):
+
+- Implement the relevant methods (or just do ``pass`` if you do not
+  need to do anything in the particular method). An adapter should not
+  use parts of OpenSCM other than ``ParameterSet``.
```

Comment:
To check I understand, the adapter only ever communicates with ParameterSet? Either as a place to retrieve input from, or a place to write output to?

Reply:
yes

Reply:
How would you make that clearer in the docs here?

Reply:
'The only part of OpenSCM with which Adapters should interact is ``ParameterSet``.'